Sources/Fuzzilli/FuzzIL/TypeSystem.swift

```diff
@@ -1007,7 +1054,7 @@ extension FunctionSignature: ProtobufConvertible {
     }
 
     init(from proto: ProtobufType, with typeCache: TypeCache?) throws {
-        self.init(expects: try proto.inputTypes.map({ try Type(from: $0, with: typeCache) }),
+        self.init(expects: try proto.parameters.map({ try Parameter(from: $0) }),
```

Comment:
Do you need to pass the `typeCache` here into the Parameter constructor as well?

Reply:
nice! yeah you're right and this is the solution to the `typeCache` serialisation bug that I was trying to work out.